src/main/java/com/twilio/http/Request.java

```diff
@@ -141,10 +146,42 @@ public URL constructURL() {
         }
     }
 
+    private String buildURL() {
+        if (region != null || edge != null) {
+            try {
+                final URL parsedUrl = new URL(url);
+                final String[] pieces = parsedUrl.getHost().split("\\.");
+                final String product = pieces[0];
```

Comment:
nit - might be good to hold the other `pieces` in nicely named variables.

Reply:
That's what L160-165 do below.

Reply:
Ah, yes. How about `pieces[pieces.length - 2]`?

Reply:
Meh, we haven't done that elsewhere.